articles/article-088-no-free-lunch.md

```diff
@@ -0,0 +1,25 @@
+<!-- BEGIN TITLE -->
+# There is no such thing as a free lunch
+<!-- END TITLE -->
+
+<!-- BEGIN BODY -->
+In Computer Science we have  a wide range of very different problems and accordingly many approaches to tackle them.
+We might for example try to find some 'optimal' solution for a search problem, but it is unclear which specific algorithm one should use.
+Hence, it is only natural to think about algorithms that could always provide us the best solution, regardless of the problems subtleties.
```

Comment:
Hence, it is natural to think about the algorithm that always provides the best solution. 
- I think problem subtleties can make or break what algorithm you want to use for instance if you want to solve the traveling salesman problem vs. if there exists a path from A -> B

Reply:
Yes, that's the whole point of the NFT ;).

Blindly throwing an method on a black-box problem won't solve anything, since "any two algorithms are equivalent when their performance is averaged across all possible problems."

The argument here is that we (need to) include some form of prior knowledge about the problem when designing the algorithm for a given problem.

Reply:
Okay I see what you are saying. The added  'could' implies that the suggested solution maybe a misnomer. 
Hence, it is natural for us to choose algorithms that we think provides the best solution, regardless of the problem subtitles. I just made it more clear that we make these assumptions.